test/functional/tool_wallet.py

```diff
@@ -36,8 +42,32 @@ def assert_tool_output(self, output, *args):
         assert_equal(stderr, '')
         assert_equal(stdout, output)
 
-    def run_test(self):
+    def wallet_path(self):
+        return os.path.join(self.nodes[0].datadir, 'regtest', 'wallets', 'wallet.dat')
+
+    def wallet_shasum(self):
+        h = hashlib.sha1()
+        mv = memoryview(bytearray(BUFFER_SIZE))
+        with open(self.wallet_path(), 'rb', buffering=0) as f:
+            for n in iter(lambda : f.readinto(mv), 0):
+                h.update(mv[:n])
+        return h.hexdigest()
+
+    def wallet_timestamp(self, op):
+        timestamp = os.path.getmtime(self.wallet_path())
+        self.log.info('Wallet file timestamp {0}: {1}'.format(op, timestamp))
```

Comment:
style-nit: Is it required to log the timestamp in a simple getter method? I'd prefer if it was logged by the caller or just asserted (Looks like you already do both of that)

Reply:
Done.